<a href="https://colab.research.google.com/github/omyahro/Data_200_Royals/blob/main/Assignment1_Food_Access.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Do low-income, predominantly Black communities experience greater food access barriers than other racial groups?**

Supermarket access, economic conditions, and food assistance affect food availability across racial groups, highlighting inequalities in food security. Studying these factors can help address disparities and improve food access in underserved communities. The findings will highlight the disproportionate impact of food insecurity on Black communities and emphasize the need for policy interventions to address racial disparities in food access.




# **Problem Statement:**

Low-income, predominantly Black communities often experience significant barriers to accessing affordable and nutritious food. Many of these neighborhoods are classified as food deserts, where grocery stores are scarce, and fresh food options are limited. As a result, residents frequently rely on convenience stores and fast food, leading to higher rates of food insecurity and diet-related health issues.


# **Data Definition:**

The Food Access Research Atlas provides a geographic analysis of food accessibility in low-income and other census tracts using various measures of supermarket proximity. It includes data on food access for populations within these tracts and offers census-tract-level information that can be used for research or community planning.


https://catalog.data.gov/dataset/food-access-research-atlas


In [5]:
# Import the libraries
import numpy as np                  # Scientific Computing
import pandas as pd                 # Data Analysis
import matplotlib.pyplot as plt     # Plotting
import seaborn as sns               # Statistical Data Visualization

# Let's make sure pandas returns all the rows and columns for the dataframe
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

# Force pandas to display full numbers instead of scientific notation
# pd.options.display.float_format = '{:.0f}'.format

# Library to suppress warnings
import warnings
warnings.filterwarnings('ignore')

In [6]:
from google.colab import drive

# Mount Google Drive
drive.mount('/content/drive')

# Define the file path (adjust to your folder location)
file_path = "/content/drive/My Drive/Food_Access_Research_Atlas.csv"

# Read the Excel file
food_access = pd.read_csv(file_path)



Mounted at /content/drive


In [7]:
# Display the first seven rows of the dataframe
food_access.head(7)

CensusTract    State          County  Urban  Pop2010  OHU2010  \
0   1001020100  Alabama  Autauga County      1     1912      693   
1   1001020200  Alabama  Autauga County      1     2170      743   
2   1001020300  Alabama  Autauga County      1     3373     1256   
3   1001020400  Alabama  Autauga County      1     4386     1722   
4   1001020500  Alabama  Autauga County      1    10766     4082   
5   1001020600  Alabama  Autauga County      1     3668     1311   
6   1001020700  Alabama  Autauga County      1     2891     1188   

   GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  \
0                  0       0.0      0.00                  0   
1                  0     181.0      8.34                  1   
2                  0       0.0      0.00                  0   
3                  0       0.0      0.00                  0   
4                  0     181.0      1.68                  0   
5                  0       0.0      0.00                  1   
6                  0      36.0      1.25                  1   

   LILATracts_halfAnd10  LILATracts_1And20  LILATracts_Vehicle  HUNVFlag  \
0                     0                  0                   0         0   
1                     1                  1                   0         0   
2                     0                  0                   0         0   
3                     0                  0                   0         0   
4                     0                  0                   0         1   
5                     1                  1                   0         0   
6                     1                  1                   0         0   

   LowIncomeTracts  PovertyRate  MedianFamilyIncome  LA1and10  LAhalfand10  \
0                0         11.3             81250.0         1            1   
1                1         17.9             49000.0         1            1   
2                0         15.0             62609.0         1            1   
3                0          2.8             70607.0         1            1   
4                0         15.2             96334.0         1            1   
5                1         21.6             69521.0         1            1   
6                1         30.5             39875.0         1            1   

   LA1and20  LATracts_half  LATracts1  LATracts10  LATracts20  \
0         1              1          1           0           0   
1         1              1          1           0           0   
2         1              1          1           0           0   
3         1              1          1           0           0   
4         1              1          1           0           0   
5         1              1          1           0           0   
6         1              1          1           0           0   

   LATractsVehicle_20  LAPOP1_10  LAPOP05_10  LAPOP1_20  LALOWI1_10  \
0                   0     1896.0      1912.0     1896.0       461.0   
1                   0     1261.0      2170.0     1261.0       604.0   
2                   0     1552.0      2857.0     1552.0       478.0   
3                   0     1363.0      3651.0     1363.0       343.0   
4                   1     2643.0      7778.0     2643.0       586.0   
5                   0     3438.0      3668.0     3438.0      1585.0   
6                   0     1231.0      2287.0     1231.0       742.0   

   LALOWI05_10  LALOWI1_20  lapophalf  lapophalfshare  lalowihalf  \
0        467.0       461.0     1912.0          100.00       467.0   
1        962.0       604.0     2170.0          100.00       962.0   
2        971.0       478.0     2857.0           84.70       971.0   
3        893.0       343.0     3651.0           83.24       893.0   
4       1719.0       586.0     7778.0           72.25      1719.0   
5       1674.0      1585.0     3668.0          100.00      1674.0   
6       1307.0       742.0     2287.0           79.12      1307.0   

   lalowihalfshare  lakidshalf  lakidshalfshare  laseniorshalf  \
0         

In [8]:
# Display the bottom seven rows of the dataframe
food_access.tail(7)


CensusTract    State           County  Urban  Pop2010  OHU2010  \
72524  56041975300  Wyoming     Uinta County      0     7761     2696   
72525  56041975400  Wyoming     Uinta County      0     6852     2632   
72526  56043000200  Wyoming  Washakie County      0     3326     1317   
72527  56043000301  Wyoming  Washakie County      1     2665     1154   
72528  56043000302  Wyoming  Washakie County      1     2542     1021   
72529  56045951100  Wyoming    Weston County      0     3314     1322   
72530  56045951300  Wyoming    Weston County      1     3894     1699   

       GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  \
72524                  0     205.0      2.64                  0   
72525                  0      65.0      0.95                  0   
72526                  0      57.0      1.71                  0   
72527                  0      10.0      0.38                  0   
72528                  0      73.0      2.87                  0   
72529                  0     252.0      7.60                  0   
72530                  0      61.0      1.57                  0   

       LILATracts_halfAnd10  LILATracts_1And20  LILATracts_Vehicle  HUNVFlag  \
72524                     0                  0                   0         0   
72525                     0                  0                   1         1   
72526                     0                  0                   0         0   
72527                     0                  0                   0         0   
72528                     0                  0                   0         0   
72529                     0                  0                   0         0   
72530                     0                  0                   0         0   

       LowIncomeTracts  PovertyRate  MedianFamilyIncome  LA1and10  \
72524                0         13.6             62445.0         0   
72525                1         17.3             57248.0         0   
72526                0          9.7             67254.0         1   
72527                0         11.6             64152.0         0   
72528                0         16.3             69605.0         0   
72529                0         17.5             74500.0         1   
72530                0         17.3             76838.0         1   

       LAhalfand10  LA1and20  LATracts_half  LATracts1  LATracts10  \
72524            0         0              0          0           0   
72525            0         0              0          0           0   
72526            1         1              0          0           1   
72527            1         0              1          0           0   
72528            1         0              1          0           0   
72529            1         0              0          0           1   
72530            1         1              1          1           0   

       LATracts20  LATractsVehicle_20  LAPOP1_10  LAPOP05_10  LAPOP1_20  \
72524           0                   0      479.0       479.0        5.0   
72525           0                   1      120.0       120.0        NaN   
72526           1                   1      902.0       902.0      730.0   
72527           0                   0        NaN       743.0        NaN   
72528           0                   0      155.0       892.0      155.0   
72529           0                   0      840.0       840.0      188.0   
72530           0                   0     1376.0      2528.0     1376.0   

       LALOWI1_10  LALOWI05_10  LALOWI1_20  lapophalf  lapophalfshare  \
72524       264.0        264.0         3.0     6037.0           77.78   
72525        24.0         24.0         NaN     5809.0           84.78   
72526       218.0        218.0       176.0     3326.0           99.99   
72527         NaN        279.0         NaN      743.0           27.89   
72528        44.0        261.0        44.0      892.0           35.10   
72529       247.0        247.0        55.0     2709.0           81.74   
72530       384.0 

In [9]:
food_access.shape


(72531, 147)

In [10]:
72531 * 147

10662057

State the shape of the dataframe :

How many rows does the dataframe have?

- 72531 rows

How many columns does the dataframe have?

- 147 columns

What is the total number of datapoints expected in the dataset (rows x columns)?

- 10662057 total datapoints

In [11]:
food_access.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72531 entries, 0 to 72530
Columns: 147 entries, CensusTract to TractSNAP
dtypes: float64(126), int64(19), object(2)
memory usage: 81.3+ MB


In [12]:
food_access.info(show_counts=True, verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72531 entries, 0 to 72530
Data columns (total 147 columns):
 #    Column                Non-Null Count  Dtype  
---   ------                --------------  -----  
 0    CensusTract           72531 non-null  int64  
 1    State                 72531 non-null  object 
 2    County                72531 non-null  object 
 3    Urban                 72531 non-null  int64  
 4    Pop2010               72531 non-null  int64  
 5    OHU2010               72531 non-null  int64  
 6    GroupQuartersFlag     72531 non-null  int64  
 7    NUMGQTRS              72506 non-null  float64
 8    PCTGQTRS              72506 non-null  float64
 9    LILATracts_1And10     72531 non-null  int64  
 10   LILATracts_halfAnd10  72531 non-null  int64  
 11   LILATracts_1And20     72531 non-null  int64  
 12   LILATracts_Vehicle    72531 non-null  int64  
 13   HUNVFlag              72531 non-null  int64  
 14   LowIncomeTracts       72531 non-null  int64  
 15   

In [13]:
food_accessColumns = food_access.columns
food_accessColumns

Index(['CensusTract', 'State', 'County', 'Urban', 'Pop2010', 'OHU2010',
       'GroupQuartersFlag', 'NUMGQTRS', 'PCTGQTRS', 'LILATracts_1And10',
       ...
       'TractSeniors', 'TractWhite', 'TractBlack', 'TractAsian', 'TractNHOPI',
       'TractAIAN', 'TractOMultir', 'TractHispanic', 'TractHUNV', 'TractSNAP'],
      dtype='object', length=147)

In [14]:
food_access.isnull().sum()


CensusTract                 0
State                       0
County                      0
Urban                       0
Pop2010                     0
OHU2010                     0
GroupQuartersFlag           0
NUMGQTRS                   25
PCTGQTRS                   25
LILATracts_1And10           0
LILATracts_halfAnd10        0
LILATracts_1And20           0
LILATracts_Vehicle          0
HUNVFlag                    0
LowIncomeTracts             0
PovertyRate                 3
MedianFamilyIncome        748
LA1and10                    0
LAhalfand10                 0
LA1and20                    0
LATracts_half               0
LATracts1                   0
LATracts10                  0
LATracts20                  0
LATractsVehicle_20          0
LAPOP1_10               29957
LAPOP05_10              14540
LAPOP1_20               35914
LALOWI1_10              29957
LALOWI05_10             14540
LALOWI1_20              35914
lapophalf                4568
lapophalfshare           4568
lalowihalf               4568
lalowihalfshare          4568
lakidshalf               4568
lakidshalfshare          4568
laseniorshalf            4568
laseniorshalfshare       4568
lawhitehalf              4568
lawhitehalfshare         4568
lablackhalf              4568
lablackhalfshare         4568
laasianhalf              4568
laasianhalfshare         4568
lanhopihalf              4568
lanhopihalfshare         4568
laaianhalf               4568
laaianhalfshare          4568
laomultirhalf            4568
laomultirhalfshare       4568
lahisphalf               4568
lahisphalfshare          4568
lahunvhalf               4568
lahunvhalfshare          4562
lasnaphalf               4568
lasnaphalfshare          4562
lapop1                  19989
lapop1share             19989
lalowi1                 19989
lalowi1share            19989
lakids1                 19989
lakids1share            19989
laseniors1              19989
laseniors1share         19989
lawhite1                19989
lawhite1share           19989
lablack1                19989
lablack1share           19989
laasian1                19989
laasian1share           19989
lanhopi1                19989
lanhopi1share           19989
laaian1                 19989
laaian1share            19989
laomultir1              19989
laomultir1share         19989
lahisp1                 19989
lahisp1share            19989
lahunv1                 19989
lahunv1share            19966
lasnap1                 19989
lasnap1share            19966
lapop10                 64765
lapop10share            64765
lalowi10                64765
lalowi10share           64765
lakids10                64765
lakids10share           64765
laseniors10             64765
laseniors10share        64765
lawhite10               64765
lawhite10share          64765
lablack10               64765
lablack10share          64765
laasian10               64765
laasian10share          64765
lanhopi10               64765
lanhopi10share          64765
laaian10                64765
laaian10share           64765
laomultir10             64765
laomultir10share        64765
lahisp10                64765
lahisp10share           64765
lahunv10                64765
lahunv10share           64666
lasnap10                64765
lasnap10share           64666
lapop20                 71025
lapop20share            71025
lalowi20                71025
lalowi20share           71025
lakids20                71025
lakids20share           71025
laseniors20             71025
laseniors20share        71025
lawhite20               71025
lawhite20share          71025
lablack20               71025
lablack20share          71025
laasian20               71025
laasian20share          71025
lanhopi20               71025
lanhopi20share          71025
laaian20                71025
laaian20share           71025
laomultir20             71025
laomultir20share        71025
lahisp20                71025
lahisp20share           71025
lahunv20                71025
lahunv20share           70920
lasnap20  

In [15]:
def missing(DataFrame):
    print ('Percentage of missing values in the dataset:\n',
           round((DataFrame.isnull().sum() * 100/ len(DataFrame)),2).sort_values(ascending=False))


# Call the function and execute
# Syntax: missing(DataFrame)
missing(food_access)

Percentage of missing values in the dataset:
 lanhopi20               97.92
lalowi20                97.92
lanhopi20share          97.92
laaian20                97.92
laaian20share           97.92
laomultir20             97.92
laomultir20share        97.92
lahisp20                97.92
lahunv20                97.92
lasnap20                97.92
lapop20                 97.92
lapop20share            97.92
lahisp20share           97.92
lalowi20share           97.92
lawhite20               97.92
lakids20                97.92
laasian20               97.92
lablack20share          97.92
lablack20               97.92
lawhite20share          97.92
laasian20share          97.92
laseniors20             97.92
lakids20share           97.92
laseniors20share        97.92
lahunv20share           97.78
lasnap20share           97.78
laasian10share          89.29
laseniors10             89.29
lablack10share          89.29
lablack10               89.29
lawhite10share          89.29
lawhite10               

**Observations**:

How many missing values are there?

- There are missing values in many of the columns; some variables have missing data of over 97%, while others have moderate missingness of between 27% and 49%.  Some of the columns, such as State, County, and CensusTract, have no missing data, while others have minimal missing values, (<1%).


Are these concentrated in specific rows or columns? How does this affect the analysis?

- The majority of missing values are found in specific columns, particularly those pertaining to race and food access.  Comparing various racial groups may become more difficult as a result, and bias may develop if more data is missing from some communities or regions.

Based on the information that is present, how should the missing values be handled?
How will this affect the analysis?
- Columns missing too much data (over 30%) should be removed. Those with moderate missing data (5%-20%) can be filled in using averages or the most common values. Removing extremely incomplete columns guarantees more trustworthy results.  Although it helps preserve more data, filling in missing values could result in minor mistakes.  The analysis may understate disparities in food access in some communities if missing data is dispersed unevenly.





In [16]:
columns_to_drop = [
    'lanhopi20', 'lanhopi20share', 'laaian20', 'laaian20share', 'laomultir20', 'laomultir20share',
    'lahisp20', 'lahisp20share', 'lawhite20', 'lawhite20share', 'laasian20', 'laasian20share',
    'laseniors20', 'laseniors20share', 'lakids20', 'lakids20share', 'lapop20', 'lapop20share',
    'lalowi20', 'lalowi20share', 'lahunv20', 'lahunv20share', 'lasnap20', 'lasnap20share',
    'laasian10', 'laasian10share', 'lahisp10', 'lahisp10share', 'lablack10', 'lablack10share',
    'lawhite10', 'lawhite10share', 'laseniors10', 'laseniors10share', 'lakids10', 'lakids10share',
    'lalowi10', 'lalowi10share', 'lapop10', 'lapop10share', 'lahunv10', 'lahunv10share',
    'lasnap10', 'lasnap10share', 'LAPOP1_20', 'LALOWI1_20', 'LALOWI1_10', 'LAPOP1_10'
]

food_access = food_access.drop(columns=[col for col in columns_to_drop if col in food_access.columns], errors='ignore')

# Check missing values
food_access.isnull().sum()



CensusTract                 0
State                       0
County                      0
Urban                       0
Pop2010                     0
OHU2010                     0
GroupQuartersFlag           0
NUMGQTRS                   25
PCTGQTRS                   25
LILATracts_1And10           0
LILATracts_halfAnd10        0
LILATracts_1And20           0
LILATracts_Vehicle          0
HUNVFlag                    0
LowIncomeTracts             0
PovertyRate                 3
MedianFamilyIncome        748
LA1and10                    0
LAhalfand10                 0
LA1and20                    0
LATracts_half               0
LATracts1                   0
LATracts10                  0
LATracts20                  0
LATractsVehicle_20          0
LAPOP05_10              14540
LALOWI05_10             14540
lapophalf                4568
lapophalfshare           4568
lalowihalf               4568
lalowihalfshare          4568
lakidshalf               4568
lakidshalfshare          4568
laseniorshalf            4568
laseniorshalfshare       4568
lawhitehalf              4568
lawhitehalfshare         4568
lablackhalf              4568
lablackhalfshare         4568
laasianhalf              4568
laasianhalfshare         4568
lanhopihalf              4568
lanhopihalfshare         4568
laaianhalf               4568
laaianhalfshare          4568
laomultirhalf            4568
laomultirhalfshare       4568
lahisphalf               4568
lahisphalfshare          4568
lahunvhalf               4568
lahunvhalfshare          4562
lasnaphalf               4568
lasnaphalfshare          4562
lapop1                  19989
lapop1share             19989
lalowi1                 19989
lalowi1share            19989
lakids1                 19989
lakids1share            19989
laseniors1              19989
laseniors1share         19989
lawhite1                19989
lawhite1share           19989
lablack1                19989
lablack1share           19989
laasian1                19989
laasian1share           19989
lanhopi1                19989
lanhopi1share           19989
laaian1                 19989
laaian1share            19989
laomultir1              19989
laomultir1share         19989
lahisp1                 19989
lahisp1share            19989
lahunv1                 19989
lahunv1share            19966
lasnap1                 19989
lasnap1share            19966
lanhopi10               64765
lanhopi10share          64765
laaian10                64765
laaian10share           64765
laomultir10             64765
laomultir10share        64765
lablack20               71025
lablack20share          71025
TractLOWI                   4
TractKids                   4
TractSeniors                4
TractWhite                  4
TractBlack                  4
TractAsian                  4
TractNHOPI                  4
TractAIAN                   4
TractOMultir                4
TractHispanic               4
TractHUNV                   4
TractSNAP                   4
dtype: int64

In [17]:
food_access.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72531 entries, 0 to 72530
Data columns (total 99 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   CensusTract           72531 non-null  int64  
 1   State                 72531 non-null  object 
 2   County                72531 non-null  object 
 3   Urban                 72531 non-null  int64  
 4   Pop2010               72531 non-null  int64  
 5   OHU2010               72531 non-null  int64  
 6   GroupQuartersFlag     72531 non-null  int64  
 7   NUMGQTRS              72506 non-null  float64
 8   PCTGQTRS              72506 non-null  float64
 9   LILATracts_1And10     72531 non-null  int64  
 10  LILATracts_halfAnd10  72531 non-null  int64  
 11  LILATracts_1And20     72531 non-null  int64  
 12  LILATracts_Vehicle    72531 non-null  int64  
 13  HUNVFlag              72531 non-null  int64  
 14  LowIncomeTracts       72531 non-null  int64  
 15  PovertyRate        

In [18]:
food_access.describe()

CensusTract         Urban       Pop2010       OHU2010  \
count  7.253100e+04  72531.000000  72531.000000  72531.000000   
mean   2.782573e+10      0.760626   4256.739022   1609.191821   
std    1.581647e+10      0.426704   1955.987626    725.676046   
min    1.001020e+09      0.000000      1.000000      0.000000   
25%    1.212708e+10      1.000000   2899.000000   1108.000000   
50%    2.712979e+10      1.000000   4011.000000   1525.000000   
75%    4.103900e+10      1.000000   5330.500000   2021.000000   
max    5.604595e+10      1.000000  37452.000000  16043.000000   

       GroupQuartersFlag      NUMGQTRS      PCTGQTRS  LILATracts_1And10  \
count       72531.000000  72506.000000  72506.000000       72531.000000   
mean            0.007114    110.121549      2.708677           0.128125   
std             0.084046    443.931753      9.570875           0.334231   
min             0.000000      0.000000      0.000000           0.000000   
25%             0.000000      0.000000      0.000000           0.000000   
50%             0.000000      7.000000      0.180000           0.000000   
75%             0.000000     64.000000      1.570000           0.000000   
max             1.000000  19496.000000    100.000000           1.000000   

       LILATracts_halfAnd10  LILATracts_1And20  LILATracts_Vehicle  \
count          72531.000000       72531.000000        72531.000000   
mean               0.279150           0.112228            0.139609   
std                0.448584           0.315649            0.346584   
min                0.000000           0.000000            0.000000   
25%                0.000000           0.000000            0.000000   
50%                0.000000           0.000000            0.000000   
75%                1.000000           0.000000            0.000000   
max                1.000000           1.000000            1.000000   

           HUNVFlag  LowIncomeTracts   PovertyRate  MedianFamilyIncome  \
count  72531.000000     72531.000000  72528.000000        71783.000000   
mean       0.210820         0.417573     15.183864        77037.792249   
std        0.407894         0.493162     11.919903        37544.445885   
min        0.000000         0.000000      0.000000         2499.000000   
25%        0.000000         0.000000      6.500000        51484.000000   
50%        0.000000         0.000000     12.000000        68821.000000   
75%        0.000000         1.000000     20.600000        93868.500000   
max        1.000000         1.000000    100.000000       250001.000000   

           LA1and10   LAhalfand10      LA1and20  LATracts_half     LATracts1  \
count  72531.000000  72531.000000  72531.000000   72531.000000  72531.000000   
mean       0.379810      0.682756      0.340668       0.638830      0.335884   
std        0.485343      0.465406      0.473937       0.480343      0.472302   
min        0.000000      0.000000      0.000000       0.000000      0.000000   
25%        0.000000      0.000000      0.000000       0.000000      0.000000   
50%        0.000000      1.000000      0.000000       1.000000      0.000000   
75%        1.000000      1.000000      1.000000       1.000000      1.000000   
max        1.000000      1.000000      1.000000       1.000000      1.000000   

         LATracts10    LATracts20  LATractsVehicle_20    LAPOP05_10  \
count  72531.000000  72531.000000        72531.000000  57991.000000   
mean       0.043926      0.004784            0.214750   2657.206946   
std        0.204932      0.069002            0.410651   2044.331313   
min        0.000000      0.000000            0.000000      0.000000   
25%        0.000000      0.000000            0.000000   1083.000000   
50%        0.000000      0.000000            0.000000   2387.000000   
75%        0.000000      0.000000            0.000000   3827.000000   
max        1.000000      1.000000            1.000000  32582.000000   

        LALOWI05_10     lapophalf  lapophalfshare    lalowihalf  \
count  57991.000000 

In [19]:
maxValuesFA = food_access.max()

maxValuesFA


CensusTract                56045951300
State                          Wyoming
County                  Ziebach County
Urban                                1
Pop2010                          37452
OHU2010                          16043
GroupQuartersFlag                    1
NUMGQTRS                       19496.0
PCTGQTRS                         100.0
LILATracts_1And10                    1
LILATracts_halfAnd10                 1
LILATracts_1And20                    1
LILATracts_Vehicle                   1
HUNVFlag                             1
LowIncomeTracts                      1
PovertyRate                      100.0
MedianFamilyIncome            250001.0
LA1and10                             1
LAhalfand10                          1
LA1and20                             1
LATracts_half                        1
LATracts1                            1
LATracts10                           1
LATracts20                           1
LATractsVehicle_20                   1
LAPOP05_10                     32582.0
LALOWI05_10                     9874.0
lapophalf                      37452.0
lapophalfshare                   100.0
lalowihalf                     19602.0
lalowihalfshare                  100.0
lakidshalf                      9084.0
lakidshalfshare                   90.8
laseniorshalf                  15261.0
laseniorshalfshare               100.0
lawhitehalf                    28477.0
lawhitehalfshare                 100.0
lablackhalf                    13594.0
lablackhalfshare                 100.0
laasianhalf                     6964.0
laasianhalfshare                 100.0
lanhopihalf                     2786.0
lanhopihalfshare                 85.88
laaianhalf                      8507.0
laaianhalfshare                  100.0
laomultirhalf                   6415.0
laomultirhalfshare               100.0
lahisphalf                     12805.0
lahisphalfshare                  100.0
lahunvhalf                      1803.0
lahunvhalfshare                  100.0
lasnaphalf                      1582.0
lasnaphalfshare                  100.0
lapop1                         37061.0
lapop1share                      100.0
lalowi1                        19397.0
lalowi1share                     100.0
lakids1                         8907.0
lakids1share                      90.8
laseniors1                     10349.0
laseniors1share                  100.0
lawhite1                       28165.0
lawhite1share                    100.0
lablack1                       12112.0
lablack1share                    100.0
laasian1                        5809.0
laasian1share                    100.0
lanhopi1                        2164.0
lanhopi1share                    85.88
laaian1                         8444.0
laaian1share                     100.0
laomultir1                      6146.0
laomultir1share                  100.0
lahisp1                        11502.0
lahisp1share                     100.0
lahunv1                         1794.0
lahunv1share                     100.0
lasnap1                         1582.0
lasnap1share                     100.0
lanhopi10                        266.0
lanhopi10share                   85.88
laaian10                        6947.0
laaian10share                    99.34
laomultir10                     1724.0
laomultir10share                 45.45
lablack20                       1086.0
lablack20share                   20.41
TractLOWI                      12562.0
TractKids                      11845.0
TractSeniors                   17271.0
TractWhite                     28983.0
TractBlack                     16804.0
TractAsian                     10485.0
TractNHOPI                      3491.0
TractAIAN                       9009.0
TractOMultir                    8839.0
TractHispanic                  15420.0
TractHUNV                       6059.0
TractSNAP                       2175.0
dtype: object

In [20]:
minValuesFA = food_access.min()
minValuesFA


CensusTract                   1001020100
State                            Alabama
County                  Abbeville County
Urban                                  0
Pop2010                                1
OHU2010                                0
GroupQuartersFlag                      0
NUMGQTRS                             0.0
PCTGQTRS                             0.0
LILATracts_1And10                      0
LILATracts_halfAnd10                   0
LILATracts_1And20                      0
LILATracts_Vehicle                     0
HUNVFlag                               0
LowIncomeTracts                        0
PovertyRate                          0.0
MedianFamilyIncome                2499.0
LA1and10                               0
LAhalfand10                            0
LA1and20                               0
LATracts_half                          0
LATracts1                              0
LATracts10                             0
LATracts20                             0
LATractsVehicle_20                     0
LAPOP05_10                           0.0
LALOWI05_10                          0.0
lapophalf                            0.0
lapophalfshare                       0.0
lalowihalf                           0.0
lalowihalfshare                      0.0
lakidshalf                           0.0
lakidshalfshare                      0.0
laseniorshalf                        0.0
laseniorshalfshare                   0.0
lawhitehalf                          0.0
lawhitehalfshare                     0.0
lablackhalf                          0.0
lablackhalfshare                     0.0
laasianhalf                          0.0
laasianhalfshare                     0.0
lanhopihalf                          0.0
lanhopihalfshare                     0.0
laaianhalf                           0.0
laaianhalfshare                      0.0
laomultirhalf                        0.0
laomultirhalfshare                   0.0
lahisphalf                           0.0
lahisphalfshare                      0.0
lahunvhalf                           0.0
lahunvhalfshare                      0.0
lasnaphalf                           0.0
lasnaphalfshare                      0.0
lapop1                               0.0
lapop1share                          0.0
lalowi1                              0.0
lalowi1share                         0.0
lakids1                              0.0
lakids1share                         0.0
laseniors1                           0.0
laseniors1share                      0.0
lawhite1                             0.0
lawhite1share                        0.0
lablack1                             0.0
lablack1share                        0.0
laasian1                             0.0
laasian1share                        0.0
lanhopi1                             0.0
lanhopi1share                        0.0
laaian1                              0.0
laaian1share                         0.0
laomultir1                           0.0
laomultir1share                      0.0
lahisp1                              0.0
lahisp1share                         0.0
lahunv1                              0.0
lahunv1share                         0.0
lasnap1                              0.0
lasnap1share                         0.0
lanhopi10                            0.0
lanhopi10share                       0.0
laaian10                             0.0
laaian10share                        0.0
laomultir10                          0.0
laomultir10share                     0.0
lablack20                            0.0
lablack20share                       0.0
TractLOWI                            0.0
TractKids                            0.0
TractSeniors                         0.0
TractWhite                           0.0
TractBlack                           0.0
TractAsian                           0.0
TractNHOPI                           0.0
TractAIAN                            0.0
TractOMultir                         0.0
TractHispanic                        0.0
TractHUNV              

In [21]:
food_access.describe().T


count          mean           std           min  \
CensusTract           72531.0  2.782573e+10  1.581647e+10  1.001020e+09   
Urban                 72531.0  7.606265e-01  4.267040e-01  0.000000e+00   
Pop2010               72531.0  4.256739e+03  1.955988e+03  1.000000e+00   
OHU2010               72531.0  1.609192e+03  7.256760e+02  0.000000e+00   
GroupQuartersFlag     72531.0  7.114199e-03  8.404573e-02  0.000000e+00   
NUMGQTRS              72506.0  1.101215e+02  4.439318e+02  0.000000e+00   
PCTGQTRS              72506.0  2.708677e+00  9.570875e+00  0.000000e+00   
LILATracts_1And10     72531.0  1.281245e-01  3.342307e-01  0.000000e+00   
LILATracts_halfAnd10  72531.0  2.791496e-01  4.485843e-01  0.000000e+00   
LILATracts_1And20     72531.0  1.122279e-01  3.156488e-01  0.000000e+00   
LILATracts_Vehicle    72531.0  1.396093e-01  3.465836e-01  0.000000e+00   
HUNVFlag              72531.0  2.108202e-01  4.078938e-01  0.000000e+00   
LowIncomeTracts       72531.0  4.175732e-01  4.931624e-01  0.000000e+00   
PovertyRate           72528.0  1.518386e+01  1.191990e+01  0.000000e+00   
MedianFamilyIncome    71783.0  7.703779e+04  3.754445e+04  2.499000e+03   
LA1and10              72531.0  3.798100e-01  4.853428e-01  0.000000e+00   
LAhalfand10           72531.0  6.827563e-01  4.654064e-01  0.000000e+00   
LA1and20              72531.0  3.406681e-01  4.739372e-01  0.000000e+00   
LATracts_half         72531.0  6.388303e-01  4.803429e-01  0.000000e+00   
LATracts1             72531.0  3.358840e-01  4.723018e-01  0.000000e+00   
LATracts10            72531.0  4.392605e-02  2.049320e-01  0.000000e+00   
LATracts20            72531.0  4.784161e-03  6.900245e-02  0.000000e+00   
LATractsVehicle_20    72531.0  2.147496e-01  4.106513e-01  0.000000e+00   
LAPOP05_10            57991.0  2.657207e+03  2.044331e+03  0.000000e+00   
LALOWI05_10           57991.0  7.974882e+02  7.714326e+02  0.000000e+00   
lapophalf             67963.0  3.166062e+03  2.003571e+03  0.000000e+00   
lapophalfshare        67963.0  7.342577e+01  2.906663e+01  0.000000e+00   
lalowihalf            67963.0  9.551535e+02  7.902997e+02  0.000000e+00   
lalowihalfshare       67963.0  2.309522e+01  1.650275e+01  0.000000e+00   
lakidshalf            67963.0  7.706095e+02  5.825632e+02  0.000000e+00   
lakidshalfshare       67963.0  1.739534e+01  8.473848e+00  0.000000e+00   
laseniorshalf         67963.0  4.229286e+02  3.282729e+02  0.000000e+00   
laseniorshalfshare    67963.0  1.032569e+01  7.330619e+00  0.000000e+00   
lawhitehalf           67963.0  2.428491e+03  1.772177e+03  0.000000e+00   
lawhitehalfshare      67963.0  5.606022e+01  3.034449e+01  0.000000e+00   
lablackhalf           67963.0  3.609503e+02  6.872588e+02  0.000000e+00   
lablackhalfshare      67963.0  9.064957e+00  1.686603e+01  0.000000e+00   
laasianhalf           67963.0  1.150718e+02  2.695013e+02  0.000000e+00   
laasianhalfshare      67963.0  2.398870e+00  4.930797e+00  0.000000e+00   
lanhopihalf           67963.0  4.783117e+00  3.392504e+01  0.000000e+00   
lanhopihalfshare      67963.0  1.048703e-01  7.763883e-01  0.000000e+00   
laaianhalf            67963.0  3.148485e+01  1.733569e+02  0.000000e+00   
laaianhalfshare       67963.0  8.068372e-01  4.574504e+00  0.000000e+00   
laomultirhalf         67963.0  2.252953e+02  3.315764e+02  0.000000e+00   
laomultirhalfshare    67963.0  4.989980e+00  6.172080e+00  0.000000e+00   
lahisphalf            67963.0  4.010741e+02  7.420223e+02  0.000000e+00   
lahisphalfshare       67963.0  8.631819e+00  1.341175e+01  0.000000e+00   
lahunvhalf            67963.0  6.949122e+01  8.066776e+01  0.000000e+00   
lahunvhalfshare       67969.0  4.684649e+00  6.011297e+00  0.000000e+00   
lasnaphalf            67963.0  1.354911e+02  1.359067e+02  0.000000e+00   
lasnaphalfshare       67969.0  8.934348e+00  9.072652e+00  0.000000e+00   
lapop1                52542.0  2.337676e+03  1.980761e+03  0.000000e+00   
lapop1share           52542.0  5.408174e+01  3.67

**Observations of Descriptive Statistics**

**What are the minimum and maximum values?**

The minimum population is 1, and the maximum is 37,452.  
The minimum poverty rate is 0%, and the maximum is 100%.  
The minimum median family income is 2,499, while the maximum $250,001.  
The minimum Black population is 0, and the maximum is 16,804.



**Are the mean and median values close to each other? If so this could indicate a normal distribution of the data. If not, this could indicate skewness in the data.
If the mean is smaller than the median the values are likely skewed left, toward the minimum value. If the mean is larger than the median then there is skewness to the right indicating more high values in the data distribution.**
- The mean is higher than the median for all the columns, which means the data is skewed to the right. This suggests there are some higher values pulling the average up.





